src/common.cpp

```diff
@@ -0,0 +1,95 @@
+#include "common.hh"
+
+#include "platform.h"
+
+#include <cmath>
+
+namespace common = acquire::sink::zarr::common;
+
+size_t
+common::bytes_of_type(const SampleType& type)
+{
+    CHECK(type < SampleTypeCount);
+    static size_t table[SampleTypeCount]; // = { 1, 2, 1, 2, 4, 2, 2, 2 };
+#define XXX(s, b) table[(s)] = (b)
+    XXX(SampleType_u8, 1);
+    XXX(SampleType_u16, 2);
+    XXX(SampleType_i8, 1);
+    XXX(SampleType_i16, 2);
+    XXX(SampleType_f32, 4);
+    XXX(SampleType_u10, 2);
+    XXX(SampleType_u12, 2);
+    XXX(SampleType_u14, 2);
+#undef XXX
+    return table[type];
+}
+
+size_t
+common::bytes_per_tile(const ImageDims& tile_shape, const SampleType& type)
+{
+    return bytes_of_type(type) * tile_shape.rows * tile_shape.cols;
+}
+
+size_t
+common::frames_per_chunk(const ImageDims& tile_shape,
+                         SampleType type,
+                         uint64_t max_bytes_per_chunk)
+{
+    auto bpt = (float)bytes_per_tile(tile_shape, type);
+    if (0 == bpt)
+        return 0;
+
+    return (size_t)std::floor((float)max_bytes_per_chunk / bpt);
+}
+
+size_t
+common::bytes_per_chunk(const ImageDims& tile_shape,
+                        const SampleType& type,
+                        uint64_t max_bytes_per_chunk)
+{
+    return bytes_per_tile(tile_shape, type) *
+           frames_per_chunk(tile_shape, type, max_bytes_per_chunk);
+}
+
+const char*
+common::sample_type_to_dtype(SampleType t)
+
+{
+    static const char* table[] = { "u1", "u2", "i1", "i2",
```

Comment:
Optional: slightly risky to look up the sampletype -> dtype correspondence by index rather than having an explicit map or switch/case and performance doesn't seem critical here but I don't feel too strongly.